discovery/service.go

```diff
@@ -478,45 +537,82 @@ func (d *Discovery) processNetworkAnnouncement(aMsg *networkMsg) bool {
 		}
 
 		if err := d.cfg.Router.AddEdge(edge); err != nil {
-			if !routing.IsError(err, routing.ErrOutdated) {
-				log.Errorf("unable to add edge: %v", err)
+			e := errors.Errorf("unable to add edge: %v", err)
+			if routing.IsError(err,
+				routing.ErrOutdated,
+				routing.ErrIgnored) {
+				log.Info(e)
 			} else {
-				log.Info("Unable to add edge: %v", err)
+				log.Error(e)
 			}
+			nMsg.err <- e
+			return nil
+		}
 
-			return false
+		// Channel announcement was successfully proceeded and know it
+		// might be broadcasted to other connected nodes if it was
+		// announcement with proof (remote).
+		if proof != nil {
+			announcements = append(announcements, msg)
 		}
 
-	// A new authenticated channel updates has arrived, this indicates
+		nMsg.err <- nil
+		return announcements
+
+	// A new authenticated channel edges has arrived, this indicates
 	// that the directional information for an already known channel has
 	// been updated.
 	case *lnwire.ChannelUpdateAnnouncement:
-		chanID := msg.ShortChannelID.ToUint64()
+		blockHeight := msg.ShortChannelID.BlockHeight
+		shortChanID := msg.ShortChannelID.ToUint64()
 
 		// If the advertised inclusionary block is beyond our knowledge
 		// of the chain tip, then we'll put the announcement in limbo
 		// to be fully verified once we advance forward in the chain.
-		if isPremature(&msg.ShortChannelID) {
-			blockHeight := msg.ShortChannelID.BlockHeight
-			log.Infof("Update announcement for chan_id=(%v), is "+
-				"premature: advertises height %v, only height "+
-				"%v is known", chanID, blockHeight,
-				d.bestHeight)
+		if isPremature(msg.ShortChannelID, 0) {
+			log.Infof("Update announcement for "+
+				"shortChanID=(%v), is premature: advertises "+
+				"height %v, only height %v is known",
+				shortChanID, blockHeight, d.bestHeight)
 
 			d.prematureAnnouncements[blockHeight] = append(
 				d.prematureAnnouncements[blockHeight],
-				aMsg,
+				nMsg,
 			)
-			return false
+			return nil
 		}
 
-		if aMsg.isRemote {
-			// TODO(andrew.shvv) Add update channel validation
+		// Get the node pub key as far as we don't have it in
+		// channel update announcement message and verify
+		// message signature.
+		chanInfo, _, _, err := d.cfg.Router.GetChannelByID(msg.ShortChannelID)
+		if err != nil {
+			d.announcementsToReprocess = append(
+				d.announcementsToReprocess,
+				nMsg,
+			)
+			log.Infof("unable to validate "+
+				"channel update: unable to get "+
+				"channel by shortChanID=%v: %v, "+
+				"adding message to the reprocessing "+
+				"batch", shortChanID, err)
+			return nil
 		}
 
+		var oppositePubKey *btcec.PublicKey
+		switch msg.Flags {
+		case 0:
+			oppositePubKey = chanInfo.NodeKey2
+		case 1:
+			oppositePubKey = chanInfo.NodeKey1
+		}
+
+		// TODO(andrew.shvv) Add validation
```

Comment:
What additional validation is needed here? If it's addressed in the next commit (I'm going through these in order), then this comment can safely be ignored.